### PR TITLE
remove closed session from authenticated map

### PR DESCRIPTION
**Pull request by davutcavdar**
*remove closed session from authenticated map

### CODE REVIEW

File: src/main/java/io/antmedia/security/TokenService.java [hunk 1]
```diff
@@ -102,14 +107,42 @@ public void setSettings(AppSettings settings) {
...
+
+
+	@Override
+	public void sessionDestroyed(HttpSessionEvent se) {
+		logger.info("session closed {}", se.getSession().getId());
+		
+		
+		if(!authenticatedMap.isEmpty() && authenticatedMap.containsKey(se.getSession().getId())) {
+			
+			authenticatedMap.remove(se.getSession().getId());
```
Comment (mekya) [inline src/main/java/io/antmedia/security/TokenService.java]: what if we directly  call `authenticatedMap.remove(se.getSession().getId());`  and remove the if statement.

File: src/main/java/io/antmedia/security/TokenService.java [hunk 2]
```diff
@@ -102,14 +107,42 @@ public void setSettings(AppSettings settings) {
...
+		return authenticatedMap;
+	}
+
+	public void setAuthenticatedMap(Map<String, String> authenticatedMap) {
+		this.authenticatedMap = authenticatedMap;
+	}
+
+	@Override
+	public void sessionCreated(HttpSessionEvent se) {
+		logger.info("session created {}", se.getSession().getId());
```
Comment (mekya) [inline src/main/java/io/antmedia/security/TokenService.java]: log level may be reduced to debug?

File: src/main/java/io/antmedia/security/TokenService.java [hunk 3]
```diff
@@ -14,10 +18,11 @@
 import io.antmedia.AppSettings;
 import io.antmedia.datastore.db.DataStoreFactory;
 import io.antmedia.datastore.db.IDataStore;
+import io.antmedia.datastore.db.IDataStoreFactory;
 import io.antmedia.datastore.db.types.Token;
 
-
-public class TokenService implements ApplicationContextAware, IStreamPublishSecurity{
+@WebListener
```
Comment (mekya) [inline src/main/java/io/antmedia/security/TokenService.java]: is it required?

File: src/main/java/io/antmedia/security/TokenService.java [hunk 4]
```diff
@@ -102,14 +107,42 @@ public void setSettings(AppSettings settings) {
...
+	@Override
+	public void sessionDestroyed(HttpSessionEvent se) {
+		logger.info("session closed {}", se.getSession().getId());
+		
+		
+		if(!authenticatedMap.isEmpty() && authenticatedMap.containsKey(se.getSession().getId())) {
+			
+			authenticatedMap.remove(se.getSession().getId());
+		}
+
```
Comment (mekya) [inline src/main/java/io/antmedia/security/TokenService.java]: Code indentation may be arranged?

File: src/main/java/io/antmedia/security/TokenService.java [hunk 5]
```diff
@@ -102,14 +107,42 @@ public void setSettings(AppSettings settings) {
 
 	public IDataStore getDataStore() {
 		if(dataStore == null) {
-			dataStore = ((DataStoreFactory) applicationContext.getBean("dataStoreFactory")).getDataStore();
+			dataStore = ((DataStoreFactory) applicationContext.getBean(IDataStoreFactory.BEAN_NAME)).getDataStore();
```
Comment (mekya) [inline src/main/java/io/antmedia/security/TokenService.java]: 👍

File: src/main/java/io/antmedia/security/TokenService.java (hunk 2 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/security/TokenService.java]: they are optional logs may be removed

File: src/main/java/io/antmedia/security/TokenService.java (hunk 1 above)
Comment (davutcavdar) [reply src/main/java/io/antmedia/security/TokenService.java]: ok

**Pull request closed without merging**